static/app/components/arithmeticInput/parser.tsx

```diff
@@ -49,19 +74,20 @@ function flatten(remaining: Array<Operation>): Operation | undefined {
     }
     term = nextTerm;
   }
+  // Shouldn't happen, tokenTerm checks remaining and tokenFactor should have at least 1 item
+  // This is just to help ts out
+  if (term === undefined) {
+    throw new Error('Unable to parse arithmetic');
+  }
   return term;
 }
 
-const options = {
-  TokenConverter,
-};
-
-export function parseArithmetic(query: string): Operation | null {
+export function parseArithmetic(query: string): {result: Term; error: string} {
+  const tc = new TokenConverter();
   try {
-    return grammar.parse(query, options);
-  } catch (e) {
-    // TODO(wmak): Should we capture these errors somewhere?
+    const result = grammar.parse(query, {tc});
+    return {result, error: tc.errors[0]};
```

Comment:
`tc.errors[0]` can be `undefined` if there are no errors. This would mean the types in `columnEditCollection` should either be `Map<number, string | undefined>` or `Map<number, string | null>` so we're explicit about the no error case.